Makefile.am

```diff
@@ -45,23 +47,58 @@ SOURCES = \
 	fibers/web/server.scm \
 	web/server/fibers.scm
 
-BUILT_SOURCES = fibers/config.scm
+BUILT_SOURCES = fibers/config.scm fibers/posix-clocks.scm fibers/events-impl.scm
 
 extlibdir = $(libdir)/guile/$(GUILE_EFFECTIVE_VERSION)/extensions
-extlib_LTLIBRARIES = epoll.la
 AM_CFLAGS = -I$(srcdir) $(WARN_CFLAGS) $(DEBUG_CFLAGS)
-epoll_la_SOURCES = epoll.c
-epoll_la_CFLAGS = $(AM_CFLAGS) $(GUILE_CFLAGS)
-epoll_la_LIBADD = $(GUILE_LIBS)
-epoll_la_LDFLAGS = -export-dynamic -module
-$(GOBJECTS): epoll.la
+
+extlib_LTLIBRARIES =
+
+if WITH_LIBEVENT
+extlib_LTLIBRARIES += fibers-libevent.la
+fibers_libevent_la_SOURCES = libevent.c
+fibers_libevent_la_CFLAGS = $(AM_CFLAGS) $(GUILE_CFLAGS)
+fibers_libevent_la_LDFLAGS = -module -no-undefined $(LIBEVENT_LIBS) $(GUILE_LDFLAGS)
+$(GOBJECTS): fibers-libevent.la
+
+fibers/events-impl.scm: fibers/libevent.scm
+	ln -sf libevent.scm fibers/events-impl.scm
```

Comment:
It seems they are supported.

https://www.howtogeek.com/howto/16226/complete-guide-to-symbolic-links-symlinks-on-windows-or-linux/

I'm not sure how I would do it without symbolic links. Any ideas? The problem is we do

```
(use-modules (fibers event-impl))
```

which would point to the right thing depending on the system. In C I could use `#ifdef` based on `config.h`, not sure with Guile.

Reply:
You can do 'cp' instead of 'ln'.

For a Scheme-equivalent to '#ifdef', you can do some tricks with syntax and 'include-from-path'. For example, if you do platform detection with

```
(define-module (clocks))
(eval-when (expand) ;; eval-when not strictly required, but saves some .go size
  (define-syntax platform-specific
    (lambda (s)
      #`(include-from-path #,(cond ((darwin? ((@ (system base target) target-os))))
 #'"darwin-clocks.scm")
                                                      (#t #'"generic-clocks.scm"))))))
(platform-specific)

where 'generic-clocks.scm' / 'darwin-clocks.scm' has the desired module code, except for an absent 'define-module'.

Similarly, you could implement a `(when-darwin? ...)` macro.

For feature detection, you could use AC_SUBST tricks or set an environment variable when compiling and check for that environment variable instead of `(darwin? ((@ (system base target) target-os)))`.

Reply:
Duh, a copy, of course! Interesting, I like the `include-from-path` approach. For now I'm going to leave it with just a copy, but it might be worth considering. Thank you!

Reply:
This is fixed now with `cp`.